From bbc3eea58a752a408dfa94707ce8cd7f53e6799d Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Sun, 4 Mar 2012 12:58:50 +0200
Subject: [PATCH 033/311] conf/rpc.{h,c}: new files

c2_fop_types for request/response and operations for rpc layer                   (c2_rpc_item_type_ops) are defined here.
---
 conf/rpc.c | 69 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 conf/rpc.h | 30 +++++++++++++++++++++++++++
 2 files changed, 99 insertions(+)
 create mode 100644 conf/rpc.c
 create mode 100644 conf/rpc.h

diff --git a/conf/rpc.c b/conf/rpc.c
new file mode 100644
index 0000000..9cdeadc
--- /dev/null
+++ b/conf/rpc.c
@@ -0,0 +1,69 @@
+/* -*- c -*- */
+/*
+ * COPYRIGHT 2012 XYRATEX TECHNOLOGY LIMITED
+ *
+ * THIS DRAWING/DOCUMENT, ITS SPECIFICATIONS, AND THE DATA CONTAINED
+ * HEREIN, ARE THE EXCLUSIVE PROPERTY OF XYRATEX TECHNOLOGY
+ * LIMITED, ISSUED IN STRICT CONFIDENCE AND SHALL NOT, WITHOUT
+ * THE PRIOR WRITTEN PERMISSION OF XYRATEX TECHNOLOGY LIMITED,
+ * BE REPRODUCED, COPIED, OR DISCLOSED TO A THIRD PARTY, OR
+ * USED FOR ANY PURPOSE WHATSOEVER, OR STORED IN A RETRIEVAL SYSTEM
+ * EXCEPT AS ALLOWED BY THE TERMS OF XYRATEX LICENSES AND AGREEMENTS.
+ *
+ * YOU SHOULD HAVE RECEIVED A COPY OF XYRATEX'S LICENSE ALONG WITH
+ * THIS RELEASE. IF NOT PLEASE CONTACT A XYRATEX REPRESENTATIVE
+ * http://www.xyratex.com/contact
+ *
+ * Original author: Valery V. Vorotyntsev <valery_vorotyntsev@xyratex.com>
+ * Original creation date: 04-Mar-2012
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include <config.h>
+#endif
+#define __C2_CONF_IMPL
+#include "conf/rpc.h"
+
+static const struct c2_fop_type_ops fopt_ops = {
+	/* XXX c2_xcode_fop_size_get() may be inappropriate here */
+	.fto_size_get = c2_xcode_fop_size_get
+};
+
+static struct c2_rpc_item_type_ops fetch_ops = {
+	/* return c2_xcode_length() */
+	.rito_item_size = XXX,
+	/*
+	 * Like c2_fop_item_type_default_encode(), but item_encdec()
+	 * is replaced with c2_xcode_encode().
+	 */
+	.rito_encode    = XXX,
+	/* XXX TODO should be set for confd */
+	.rito_decode    = NULL
+};
+
+/*
+ * Defines `struct c2_fop_type c2_conf_fetch_fopt'.
+ *
+ * XXX Dependency: struct c2_fop_type_format c2_conf_fetch_tfmt
+ */
+C2_FOP_TYPE_DECLARE_OPS(c2_conf_fetch, "c2_conf_fetch", &fopt_ops,
+			C2_CONF_FETCH_RESP_OPCODE, C2_RPC_ITEM_TYPE_REQUEST,
+			&fetch_ops);
+
+static struct c2_rpc_item_type_ops fetch_resp_ops = {
+	/* return c2_xcode_length() */
+	.rito_item_size = XXX,
+	/* XXX TODO should be set for confd */
+	.rito_encode    = NULL,
+	/*
+	 * - Allocate and initialize c2_fop.
+	 * - c2_xcode_decode() serialized configuration data pointed
+	 *   to by `cur' ==> dynamically allocated c2_conf_fetch_resp.
+	 * - Set c2_fop_data() to the address of c2_conf_fetch_resp.
+	 */
+	.rito_decode    = XXX
+};
+
+C2_FOP_TYPE_DECLARE_OPS(c2_conf_fetch_resp, "c2_conf_fetch_resp", &ftype_ops,
+			C2_CONF_FETCH_RESP_OPCODE, C2_RPC_ITEM_TYPE_REPLY,
+			&fetch_resp_ops);
diff --git a/conf/rpc.h b/conf/rpc.h
new file mode 100644
index 0000000..6647c91
--- /dev/null
+++ b/conf/rpc.h
@@ -0,0 +1,30 @@
+/* -*- c -*- */
+/*
+ * COPYRIGHT 2012 XYRATEX TECHNOLOGY LIMITED
+ *
+ * THIS DRAWING/DOCUMENT, ITS SPECIFICATIONS, AND THE DATA CONTAINED
+ * HEREIN, ARE THE EXCLUSIVE PROPERTY OF XYRATEX TECHNOLOGY
+ * LIMITED, ISSUED IN STRICT CONFIDENCE AND SHALL NOT, WITHOUT
+ * THE PRIOR WRITTEN PERMISSION OF XYRATEX TECHNOLOGY LIMITED,
+ * BE REPRODUCED, COPIED, OR DISCLOSED TO A THIRD PARTY, OR
+ * USED FOR ANY PURPOSE WHATSOEVER, OR STORED IN A RETRIEVAL SYSTEM
+ * EXCEPT AS ALLOWED BY THE TERMS OF XYRATEX LICENSES AND AGREEMENTS.
+ *
+ * YOU SHOULD HAVE RECEIVED A COPY OF XYRATEX'S LICENSE ALONG WITH
+ * THIS RELEASE. IF NOT PLEASE CONTACT A XYRATEX REPRESENTATIVE
+ * http://www.xyratex.com/contact
+ *
+ * Original author: Valery V. Vorotyntsev <valery_vorotyntsev@xyratex.com>
+ * Original creation date: 04-Mar-2012
+ */
+#ifndef __COLIBRI_CONF_RPC_H__
+#define __COLIBRI_CONF_RPC_H__
+
+#ifndef __C2_CONF_IMPL
+#  error "conf/rpc.h is not supposed to be #included by confc API users"
+#endif
+
+extern struct c2_fop_type c2_conf_fetch_fopt;
+extern struct c2_fop_type c2_conf_fetch_resp_fopt;
+
+#endif /* __COLIBRI_CONF_RPC_H__ */
-- 
1.8.3.2

